### PR TITLE
os/bluestore: get rid off statfs update on each txc

### PR DESCRIPTION
**Pull request by ifed01**
Instead this relies on NCB stuff to recover bluestore stats in case of non-graceful shutdown.
This functionality is a prerequisite for upcoming new WAL implementation. Additionally this might provide some performance improvements as DB gets less load.
And even more - redesigned NCB recovery is more performant too.

Signed-off-by: Igor Fedotov <igor.fedotov@croit.io>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by ifed01** [comment]
jenkins test make check

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -7897,110 +7926,95 @@ int BlueStore::_fsck_check_extents(
...
-
-      volatile_statfs vstatfs;
-      bufferlist bl = it->value();
-      auto blp = bl.cbegin();
-      try {
-	vstatfs.decode(blp);
-      } catch (ceph::buffer::error& e) {
-        derr << "fsck error: failed to decode Pool StatFS record"
-	     << pretty_binary_string(key) << dendl;
+  string key;
```
Comment (aclamk) [inline src/os/bluestore/BlueStore.cc:7936]: I feel like we can make check/repair statfs much simpler.
We could create a map of {key->stat} from the stats we just scanned.
This should be ideally reflected by state of DB.
If DB does not have something - we complain and add (in repair mode).
If DB has different value - we complain and fix (in repair mode).
If DB has extra key - we complain and delete (in repair mode).

File: src/os/bluestore/BlueStore.h [hunk 2]
```diff
@@ -210,6 +210,7 @@ enum {
 };
 
 #define META_POOL_ID ((uint64_t)-1ull)
+using bptr_c_it_t = buffer::ptr::const_iterator;
```
[the remark this answers is not in the document]
Comment (ifed01) [reply src/os/bluestore/BlueStore.h:213]: me too ;)

File: src/os/bluestore/BlueStore.h [hunk 3]
```diff
@@ -748,11 +749,11 @@ class BlueStore : public ObjectStore,
       }
     }
     void decode(
-      Collection *coll,
       ceph::buffer::ptr::const_iterator& p,
       uint64_t struct_v,
       uint64_t* sbid,
-      bool include_ref_map);
+      bool include_ref_map,
+      Collection *coll);
```
[the remark this answers is not in the document]
Comment (ifed01) [reply src/os/bluestore/BlueStore.h:756]: it's optional now hence IMO better placed at the end of the param list

File: src/os/bluestore/BlueStore.cc [hunk 4]
```diff
@@ -18675,232 +18746,245 @@ int BlueStore::restore_allocator(Allocator* dest_allocator, uint64_t *num, uint6
...
+      if (get_key_shared_blob(key, &sbid) != 0) {
+	derr << __func__ << " bad shared blob key '" << key
+	     << dendl;
+      }
+      bluestore_shared_blob_t shared_blob(sbid);
+      bufferlist bl = it->value();
+      auto blp = bl.cbegin();
+      try {
+        decode(shared_blob, blp);
+      }
```
[the remark this answers is not in the document]
Comment (ifed01) [reply src/os/bluestore/BlueStore.cc:18788]: You mean whether we  handle empty/too short bufferlist properly (i.e. .raise relevant exception)? If so then yes - it works fine with the above try..catch block - just double checked that myself.

File: src/os/bluestore/BlueStore.cc [hunk 5]
```diff
@@ -5897,21 +5900,33 @@ int BlueStore::_init_alloc(std::map<uint64_t, uint64_t> *zone_adjustments)
 
 void BlueStore::_post_init_alloc(const std::map<uint64_t, uint64_t>& zone_adjustments)
 {
+  int r = 0;
 #ifdef HAVE_LIBZBD
-  assert(bdev->is_smr());
```
[the remark this answers is not in the document]
Comment (ifed01) [reply src/os/bluestore/BlueStore.cc:5901]: I updated the invocation of this stuff - the func is not limited to SMR stuff any more.
Hence we need to distinguish srm/non-smr devices now and assert() is inappropriate.

File: src/os/bluestore/BlueStore.cc [hunk 6]
```diff
@@ -18675,232 +18746,245 @@ int BlueStore::restore_allocator(Allocator* dest_allocator, uint64_t *num, uint6
...
+        Extent le;
+        ExtentMap::decode_extent(
+          nullptr,
+          struct_v,
+          p,
+          &le,
+          pos,
+          prev_len,
+          [&](bool spanning, uint64_t blobid) {
+              auto it = spanning ?
```
[the remark this answers is not in the document]
Comment (ifed01) [reply src/os/bluestore/BlueStore.cc]: Re-implemented alternatively, you variant doesn't look perfect as well IMO ;)

File: src/os/bluestore/BlueStore.cc (hunk 1 above)
Comment (ifed01) [reply src/os/bluestore/BlueStore.cc:7949]: The tricky thing is that now we might have statfs persisted at DB in two different ways depending on the previous shutdown: graceful or not. This implementation checks in-memory stats (loaded either from relevant DB records or via recovery) vs. stats loaded by fsck.
For your suggestion we'll need to perform DB update immediately after the recovery and then de-serialize it back for fsck (and then finally update DB once again on a shutdown). Which doesn't make much sense to me...

File: src/os/bluestore/BlueStore.cc [hunk 7]
```diff
@@ -3129,82 +3130,130 @@ bool BlueStore::ExtentMap::encode_some(
   return false;
 }
 
-unsigned BlueStore::ExtentMap::decode_some(bufferlist& bl)
+void BlueStore::ExtentMap::decode_extent(
```
[the remark this answers is not in the document]
Comment (ifed01) [reply src/os/bluestore/BlueStore.cc:3133]: OK, will do the refactor and see if this is more readable

[the remark this answers is not in the document]
Comment (ifed01) [reply src/os/bluestore/BlueStore.cc]: Done.

File: src/os/bluestore/BlueStore.cc [hunk 8]
```diff
@@ -18658,228 +18752,248 @@ int BlueStore::restore_allocator(Allocator* dest_allocator, uint64_t *num, uint6
...
+    derr << "failed getting shared blob's iterator" << dendl;
+    return -ENOENT;
+  }
+  if (it) {
+    for (it->lower_bound(string()); it->valid(); it->next()) {
+      const auto& key = it->key();
+      dout(20) << __func__ << " decode sb " << pretty_binary_string(key) << dendl;
+      uint64_t sbid = 0;
+      if (get_key_shared_blob(key, &sbid) != 0) {
+	derr << __func__ << " bad shared blob key '" << pretty_binary_string(key)
```
Comment (aclamk) [inline src/os/bluestore/BlueStore.cc:18888]: only opening `'` around pretty_binary_string

File: src/os/bluestore/BlueStore.cc [hunk 9]
```diff
@@ -18494,6 +18580,14 @@ int calc_allocator_image_trailer_size()
 //-----------------------------------------------------------------------------------
 int BlueStore::__restore_allocator(Allocator* allocator, uint64_t *num, uint64_t *bytes)
 {
+  if (cct->_conf->bluestore_debug_inject_allocation_from_file_failure > 0) {
+     boost::mt11213b rng(time(NULL));
+    boost::uniform_real<> ur(0, 1);
+    if (ur(rng) < cct->_conf->bluestore_debug_inject_allocation_from_file_failure) {
```
Comment (aclamk) [inline src/os/bluestore/BlueStore.cc:18586]: I think <= is "almost" better, as it hits case when conf=1 and rng()=1

**Review by aclamk — APPROVED** [review]
Strangely, I found no errors.
I wonder how many times you recheck it yourself Igor.

File: src/os/bluestore/BlueStore.cc (hunk 9 above)
Comment (ifed01) [reply src/os/bluestore/BlueStore.cc:18586]: uniform_real returns a random floating-point value distributed in the range [min..max). Hence your case isn't valid - rng() is never equal to 1

**Comment by ifed01** [comment]
jenkins test make check

**Comment by ifed01** [comment]
jenkins test make check

**Comment by ifed01** [comment]
jenkins test make check

**Comment by ifed01** [comment]
jenkins test make check

**Comment by ifed01** [comment]
jenkins test make check

**Comment by ifed01** [comment]
jenkins test make check

**Comment by kamoltat** [comment]
Please check any of these: 
/a/yuriw-2022-07-22_03:30:40-rados-wip-yuri3-testing-2022-07-21-1604-distro-default-smithi/6943775
/a/yuriw-2022-07-22_03:30:40-rados-wip-yuri3-testing-2022-07-21-1604-distro-default-smithi/6944099

````
2022-07-22T22:44:50.061 INFO:teuthology.orchestra.run.smithi064.stdout:m(3) p(3,0-12) O(3,0-13)=block_cache={type=binned_lru} L P
2022-07-22T22:44:50.065 INFO:tasks.thrashosds.thrasher:applying new sharding to bluestore on osd.8
2022-07-22T22:44:50.066 INFO:tasks.thrashosds.thrasher:generate random sharding
2022-07-22T22:44:50.067 INFO:tasks.thrashosds.thrasher:applying new sharding: m(2) P(4,5-21) L(5)
2022-07-22T22:44:50.068 DEBUG:teuthology.orchestra.run.smithi064:> sudo ceph-bluestore-tool --err-to-stderr --no-mon-config '--log-file=/var/log/ceph/bluestore_tool.$pid.log' --log-level=10 --path /var/lib/ceph/osd/ceph-8 --sharding 'm(2) P(4,5-21) L(5)' reshard
2022-07-22T22:44:50.467 INFO:teuthology.orchestra.run.smithi064.stdout:reshard success
2022-07-22T22:44:50.469 INFO:teuthology.orchestra.run.smithi064.stderr:*** Caught signal (Segmentation fault) **
2022-07-22T22:44:50.470 INFO:teuthology.orchestra.run.smithi064.stderr: in thread 7f8b57de7740 thread_name:ceph-bluestore-
2022-07-22T22:44:50.471 INFO:teuthology.orchestra.run.smithi064.stderr: ceph version 17.0.0-13750-g19acc2f7 (19acc2f7e3edb197f028ffa801e28f62f3698c79) quincy (dev)
2022-07-22T22:44:50.471 INFO:teuthology.orchestra.run.smithi064.stderr: 1: /lib64/libpthread.so.0(+0x12ce0) [0x7f8b54ff2ce0]
2022-07-22T22:44:50.472 INFO:teuthology.orchestra.run.smithi064.stderr: 2: (RocksDBStore::submit_common(rocksdb::WriteOptions&, std::shared_ptr<KeyValueDB::TransactionImpl>)+0x7b) [0x55ddb0db841b]
2022-07-22T22:44:50.475 INFO:teuthology.orchestra.run.smithi064.stderr: 3: (RocksDBStore::submit_transaction_sync(std::shared_ptr<KeyValueDB::TransactionImpl>)+0x9a) [0x55ddb0db8e2a]
2022-07-22T22:44:50.476 INFO:teuthology.orchestra.run.smithi064.stderr: 4: (BlueStore::_close_db()+0x4e5) [0x55ddb0b8cbc5]
2022-07-22T22:44:50.477 INFO:teuthology.orchestra.run.smithi064.stderr: 5: (BlueStore::_close_db_and_around()+0x17) [0x55ddb0b8d507]
2022-07-22T22:44:50.477 INFO:teuthology.orchestra.run.smithi064.stderr: 6: (BlueStore::close_db_environment()+0xd) [0x55ddb0b8d55d]
2022-07-22T22:44:50.478 INFO:teuthology.orchestra.run.smithi064.stderr: 7: main()
2022-07-22T22:44:50.478 INFO:teuthology.orchestra.run.smithi064.stderr: 8: __libc_start_main()
2022-07-22T22:44:50.479 INFO:teuthology.orchestra.run.smithi064.stderr: 9: _start()
2022-07-22T22:44:50.480 INFO:teuthology.orchestra.run.smithi064.stderr:2022-07-22T22:44:50.463+0000 7f8b57de7740 -1 *** Caught signal (Segmentation fault) **
2022-07-22T22:44:50.481 INFO:teuthology.orchestra.run.smithi064.stderr: in thread 7f8b57de7740 thread_name:ceph-bluestore-
2022-07-22T22:44:50.481 INFO:teuthology.orchestra.run.smithi064.stderr:
2022-07-22T22:44:50.482 INFO:teuthology.orchestra.run.smithi064.stderr: ceph version 17.0.0-13750-g19acc2f7 (19acc2f7e3edb197f028ffa801e28f62f3698c79) quincy (dev)
2022-07-22T22:44:50.482 INFO:teuthology.orchestra.run.smithi064.stderr: 1: /lib64/libpthread.so.0(+0x12ce0) [0x7f8b54ff2ce0]
2022-07-22T22:44:50.483 INFO:teuthology.orchestra.run.smithi064.stderr: 2: (RocksDBStore::submit_common(rocksdb::WriteOptions&, std::shared_ptr<KeyValueDB::TransactionImpl>)+0x7b) [0x55ddb0db841b]
2022-07-22T22:44:50.483 INFO:teuthology.orchestra.run.smithi064.stderr: 3: (RocksDBStore::submit_transaction_sync(std::shared_ptr<KeyValueDB::TransactionImpl>)+0x9a) [0x55ddb0db8e2a]
2022-07-22T22:44:50.484 INFO:teuthology.orchestra.run.smithi064.stderr: 4: (BlueStore::_close_db()+0x4e5) [0x55ddb0b8cbc5]
2022-07-22T22:44:50.484 INFO:teuthology.orchestra.run.smithi064.stderr: 5: (BlueStore::_close_db_and_around()+0x17) [0x55ddb0b8d507]
2022-07-22T22:44:50.485 INFO:teuthology.orchestra.run.smithi064.stderr: 6: (BlueStore::close_db_environment()+0xd) [0x55ddb0b8d55d]
2022-07-22T22:44:50.498 INFO:teuthology.orchestra.run.smithi064.stderr: 7: main()
2022-07-22T22:44:50.498 INFO:teuthology.orchestra.run.smithi064.stderr: 8: __libc_start_main()
2022-07-22T22:44:50.499 INFO:teuthology.orchestra.run.smithi064.stderr: 9: _start()
2022-07-22T22:44:50.508 INFO:teuthology.orchestra.run.smithi064.stderr: NOTE: a copy of the executable, or `objdump -rdS <executable>` is needed to interpret this.
2022-07-22T22:44:50.508 INFO:teuthology.orchestra.run.smithi064.stderr:
2022-07-22T22:44:50.514 INFO:teuthology.orchestra.run.smithi064.stderr:     0> 2022-07-22T22:44:50.463+0000 7f8b57de7740 -1 *** Caught signal (Segmentation fault) **
2022-07-22T22:44:50.521 INFO:teuthology.orchestra.run.smithi064.stderr: in thread 7f8b57de7740 thread_name:ceph-bluestore-
2022-07-22T22:44:50.522 INFO:teuthology.orchestra.run.smithi064.stderr:
2022-07-22T22:44:50.522 INFO:teuthology.orchestra.run.smithi064.stderr: ceph version 17.0.0-13750-g19acc2f7 (19acc2f7e3edb197f028ffa801e28f62f3698c79) quincy (dev)
2022-07-22T22:44:50.524 INFO:teuthology.orchestra.run.smithi064.stderr: 1: /lib64/libpthread.so.0(+0x12ce0) [0x7f8b54ff2ce0]
2022-07-22T22:44:50.524 INFO:teuthology.orchestra.run.smithi064.stderr: 2: (RocksDBStore::submit_common(rocksdb::WriteOptions&, std::shared_ptr<KeyValueDB::TransactionImpl>)+0x7b) [0x55ddb0db841b]
2022-07-22T22:44:50.525 INFO:teuthology.orchestra.run.smithi064.stderr: 3: (RocksDBStore::submit_transaction_sync(std::shared_ptr<KeyValueDB::TransactionImpl>)+0x9a) [0x55ddb0db8e2a]
2022-07-22T22:44:50.525 INFO:teuthology.orchestra.run.smithi064.stderr: 4: (BlueStore::_close_db()+0x4e5) [0x55ddb0b8cbc5]
2022-07-22T22:44:50.525 INFO:teuthology.orchestra.run.smithi064.stderr: 5: (BlueStore::_close_db_and_around()+0x17) [0x55ddb0b8d507]
2022-07-22T22:44:50.531 INFO:teuthology.orchestra.run.smithi064.stderr: 6: (BlueStore::close_db_environment()+0xd) [0x55ddb0b8d55d]
2022-07-22T22:44:50.532 INFO:teuthology.orchestra.run.smithi064.stderr: 7: main()
2022-07-22T22:44:50.532 INFO:teuthology.orchestra.run.smithi064.stderr: 8: __libc_start_main()
2022-07-22T22:44:50.532 INFO:teuthology.orchestra.run.smithi064.stderr: 9: _start()
2022-07-22T22:44:50.532 INFO:teuthology.orchestra.run.smithi064.stderr: NOTE: a copy of the executable, or `objdump -rdS <executable>` is needed to interpret this.
2022-07-22T22:44:50.533 INFO:teuthology.orchestra.run.smithi064.stderr:
2022-07-22T22:44:50.533 INFO:teuthology.orchestra.run.smithi064.stderr:     0> 2022-07-22T22:44:50.463+0000 7f8b57de7740 -1 *** Caught signal (Segmentation fault) **
2022-07-22T22:44:50.533 INFO:teuthology.orchestra.run.smithi064.stderr: in thread 7f8b57de7740 thread_name:ceph-bluestore-
2022-07-22T22:44:50.533 INFO:teuthology.orchestra.run.smithi064.stderr:
2022-07-22T22:44:50.534 INFO:teuthology.orchestra.run.smithi064.stderr: ceph version 17.0.0-13750-g19acc2f7 (19acc2f7e3edb197f028ffa801e28f62f3698c79) quincy (dev)
2022-07-22T22:44:50.534 INFO:teuthology.orchestra.run.smithi064.stderr: 1: /lib64/libpthread.so.0(+0x12ce0) [0x7f8b54ff2ce0]
2022-07-22T22:44:50.534 INFO:teuthology.orchestra.run.smithi064.stderr: 2: (RocksDBStore::submit_common(rocksdb::WriteOptions&, std::shared_ptr<KeyValueDB::TransactionImpl>)+0x7b) [0x55ddb0db841b]
2022-07-22T22:44:50.534 INFO:teuthology.orchestra.run.smithi064.stderr: 3: (RocksDBStore::submit_transaction_sync(std::shared_ptr<KeyValueDB::TransactionImpl>)+0x9a) [0x55ddb0db8e2a]
2022-07-22T22:44:50.535 INFO:teuthology.orchestra.run.smithi064.stderr: 4: (BlueStore::_close_db()+0x4e5) [0x55ddb0b8cbc5]
2022-07-22T22:44:50.535 INFO:teuthology.orchestra.run.smithi064.stderr: 5: (BlueStore::_close_db_and_around()+0x17) [0x55ddb0b8d507]
2022-07-22T22:44:50.535 INFO:teuthology.orchestra.run.smithi064.stderr: 6: (BlueStore::close_db_environment()+0xd) [0x55ddb0b8d55d]
2022-07-22T22:44:50.535 INFO:teuthology.orchestra.run.smithi064.stderr: 7: main()
2022-07-22T22:44:50.536 INFO:teuthology.orchestra.run.smithi064.stderr: 8: __libc_start_main()
2022-07-22T22:44:50.536 INFO:teuthology.orchestra.run.smithi064.stderr: 9: _start()
2022-07-22T22:44:50.536 INFO:teuthology.orchestra.run.smithi064.stderr: NOTE: a copy of the executable, or `objdump -rdS <executable>` is needed to interpret this.
2022-07-22T22:44:50.536 INFO:teuthology.orchestra.run.smithi064.stderr:
2022-07-22T22:44:50.634 DEBUG:teuthology.orchestra.run:got remote process result: None
2022-07-22T22:44:50.636 INFO:tasks.thrashosds.thrasher:Traceback (most recent call last):
  File "/home/teuthworker/src/github.com_ceph_ceph-c_19acc2f7e3edb197f028ffa801e28f62f3698c79/qa/tasks/ceph_manager.py", line 190, in wrapper
    return func(self)
  File "/home/teuthworker/src/github.com_ceph_ceph-c_19acc2f7e3edb197f028ffa801e28f62f3698c79/qa/tasks/ceph_manager.py", line 1422, in _do_thrash
    self.choose_action()()
  File "/home/teuthworker/src/github.com_ceph_ceph-c_19acc2f7e3edb197f028ffa801e28f62f3698c79/qa/tasks/ceph_manager.py", line 1162, in test_bluestore_reshard
    self.test_bluestore_reshard_action()
  File "/home/teuthworker/src/github.com_ceph_ceph-c_19acc2f7e3edb197f028ffa801e28f62f3698c79/qa/tasks/ceph_manager.py", line 1142, in test_bluestore_reshard_action
    raise Exception("ceph-bluestore-tool resharding failed.")
Exception: ceph-bluestore-tool resharding failed.

2022-07-22T22:44:50.636 ERROR:tasks.thrashosds.thrasher:exception:
Traceback (most recent call last):
  File "/home/teuthworker/src/github.com_ceph_ceph-c_19acc2f7e3edb197f028ffa801e28f62f3698c79/qa/tasks/ceph_manager.py", line 1290, in do_thrash
    self._do_thrash()
  File "/home/teuthworker/src/github.com_ceph_ceph-c_19acc2f7e3edb197f028ffa801e28f62f3698c79/qa/tasks/ceph_manager.py", line 190, in wrapper
    return func(self)
  File "/home/teuthworker/src/github.com_ceph_ceph-c_19acc2f7e3edb197f028ffa801e28f62f3698c79/qa/tasks/ceph_manager.py", line 1422, in _do_thrash
    self.choose_action()()
  File "/home/teuthworker/src/github.com_ceph_ceph-c_19acc2f7e3edb197f028ffa801e28f62f3698c79/qa/tasks/ceph_manager.py", line 1162, in test_bluestore_reshard
    self.test_bluestore_reshard_action()
  File "/home/teuthworker/src/github.com_ceph_ceph-c_19acc2f7e3edb197f028ffa801e28f62f3698c79/qa/tasks/ceph_manager.py", line 1142, in test_bluestore_reshard_action
    raise Exception("ceph-bluestore-tool resharding failed.")
Exception: ceph-bluestore-tool resharding failed.
````

**Comment by aclamk** [comment]
@ifed01
I created temporary #47331, to address @kamoltat signalled problem.
I think we need to add https://github.com/ceph/ceph/pull/47331/commits/7b37419cf0eb35b6467abae55104567c031e81a1
either to your this PR, or make it a standalone one.

**Comment by ifed01** [comment]
> Please check any of these:
> /a/yuriw-2022-07-22_03:30:40-rados-wip-yuri3-testing-2022-07-21-1604-distro-default-smithi/6943775
> /a/yuriw-2022-07-22_03:30:40-rados-wip-yuri3-testing-2022-07-21-1604-distro-default-smithi/6944099

Looks like @aclamk fixed that. His fix is now embedded into this PR.

**Comment by ifed01** [comment]
> @ifed01 I created temporary #47331, to address @kamoltat signalled problem. I think we need to add [7b37419](https://github.com/ceph/ceph/commit/7b37419cf0eb35b6467abae55104567c031e81a1) either to your this PR, or make it a standalone one.

Done

**Comment by aclamk** [comment]
jenkins test make check

**Comment by ifed01** [comment]
jenkins test make check

**Comment by ifed01** [comment]
@aclaml - mind taking another look?

**Review by aclamk — APPROVED** [review]
I thought I already approved this.
I already have some cleanup work that depends on this one.

**Comment by sseshasa** [comment]
**Teuthology Test Result**
http://pulpito.front.sepia.ceph.com/?branch=wip-yuri5-testing-2022-08-18-0812

@ifed01 Across both the runs from the link above, the following failed jobIDs appear to be related to this PR. 

- [6981593](http://pulpito.front.sepia.ceph.com/lflores-2022-08-19_21:39:29-rados-wip-yuri5-testing-2022-08-18-0812-distro-default-smithi/6981593)
-  [6981597](http://pulpito.front.sepia.ceph.com/lflores-2022-08-19_21:39:29-rados-wip-yuri5-testing-2022-08-18-0812-distro-default-smithi/6981597)
 - [6981601](http://pulpito.front.sepia.ceph.com/lflores-2022-08-19_21:39:29-rados-wip-yuri5-testing-2022-08-18-0812-distro-default-smithi/6981601)
 - [6978863](http://pulpito.front.sepia.ceph.com/yuriw-2022-08-18_17:36:35-rados-wip-yuri5-testing-2022-08-18-0812-distro-default-smithi/6978863)
 - [6978882](http://pulpito.front.sepia.ceph.com/yuriw-2022-08-18_17:36:35-rados-wip-yuri5-testing-2022-08-18-0812-distro-default-smithi/6978882)
 - [6979092](http://pulpito.front.sepia.ceph.com/yuriw-2022-08-18_17:36:35-rados-wip-yuri5-testing-2022-08-18-0812-distro-default-smithi/6979092)

The tests are thrashosds related and in each case the failure is due to an osd not coming up after being restarted due to fsck errors being reported by _fsck_check_statfs(). Could you please take a look?

Some logs from JobID: [6978863](http://pulpito.front.sepia.ceph.com/yuriw-2022-08-18_17:36:35-rados-wip-yuri5-testing-2022-08-18-0812-distro-default-smithi/6978863)

```
...
2022-08-18T18:28:09.247+0000 7f83ac952440 10 bluefs _read h 0x5580b7c03500 0x7faa~8000 from file(ino 117 size 0xb79d mtime 2022-08-18T18:27:19.033067+0000 allocated 10000 alloc_commit 10000 extents [1:0x60000~10000]) prefetch
2022-08-18T18:28:09.247+0000 7f83ac952440 20 bluefs _read reaching (or past) eof, len clipped to 0x37f3
2022-08-18T18:28:09.247+0000 7f83ac952440 20 bluefs _read left 0x4056 len 0x37f3
2022-08-18T18:28:09.247+0000 7f83ac952440 20 bluefs _read got 14323
2022-08-18T18:28:09.248+0000 7f83902d9700 20 bluestore.MempoolThread(0x5580b7c26b38) _resize_shards cache_size: 563370167 kv_alloc: 197132288 kv_used: 204864 kv_onode_alloc: 142606336 kv_onode_used: 7497280 meta_alloc: 130023424 meta_used: 136 data_alloc: 75497472 data_used: 0
2022-08-18T18:28:09.256+0000 7f83ac952440  1 bluestore(/var/lib/ceph/osd/ceph-3) _fsck_on_open checking pool_statfs
2022-08-18T18:28:09.256+0000 7f83ac952440 -1 bluestore(/var/lib/ceph/osd/ceph-3) _fsck_check_statfs::fsck error: pool 3 has got no statfs to match against: store_statfs(0x0/0x0/0x0, data 0xeb9d000/0x975a000, compress 0x0/0x0/0x0, omap 0x0, meta 0x0)

...
...

2022-08-18T18:28:09.264+0000 7f83ac952440  1 bluestore(/var/lib/ceph/osd/ceph-3) _fsck_on_open checking deferred events
2022-08-18T18:28:09.264+0000 7f83ac952440  2 bluestore(/var/lib/ceph/osd/ceph-3) _fsck_on_open 464 objects, 0 of them sharded.
2022-08-18T18:28:09.264+0000 7f83ac952440  2 bluestore(/var/lib/ceph/osd/ceph-3) _fsck_on_open 692 extents to 692 blobs, 0 spanning, 0 shared.
2022-08-18T18:28:09.264+0000 7f83ac952440  1 bluestore(/var/lib/ceph/osd/ceph-3) _fsck_on_open <<<FINISH>>> with 1 errors, 0 warnings, 0 repaired, 1 remaining in 0.036685 seconds

...

2022-08-18T18:28:09.265+0000 7f83ac952440  1 bluefs umount
2022-08-18T18:28:09.265+0000 7f83ac952440 10 bluefs sync_metadata - no pending log events
2022-08-18T18:28:09.265+0000 7f83ac952440 10 bluefs _drain_writer 0x5580b7c40600 type 0
2022-08-18T18:28:09.265+0000 7f83ac952440 20 bluefs _stop_alloc
2022-08-18T18:28:09.265+0000 7f83ac952440  1 bdev(0x5580b89a1c00 /var/lib/ceph/osd/ceph-3/block) close
2022-08-18T18:28:09.462+0000 7f83ac952440 10 bluestore(/var/lib/ceph/osd/ceph-3) _close_fm
2022-08-18T18:28:09.462+0000 7f83ac952440  1 freelist shutdown
2022-08-18T18:28:09.462+0000 7f83ac952440  1 bdev(0x5580b89a0400 /var/lib/ceph/osd/ceph-3/block) close
2022-08-18T18:28:09.681+0000 7f83ac952440 -1 bluestore(/var/lib/ceph/osd/ceph-3) _mount fsck found 1 errors
2022-08-18T18:28:09.681+0000 7f83ac952440 -1 osd.3 0 OSD:init: unable to mount object store
2022-08-18T18:28:09.681+0000 7f83ac952440 -1 ^[[0;31m ** ERROR: osd init failed: (5) Input/output error^[[0m
```

**Comment by yuriw** [comment]
Per @sseshasa 

"@yuriweinstein I looked into the pending jobIDs (as mentioned by @nehaojha ) and all the other failures reported in the re-run (http://pulpito.front.sepia.ceph.com/lflores-2022-08-19_21:39:29-rados-wip-yuri5-testing-2022-08-18-0812-distro-default-smithi/).

Except for PR: ￼os/bluestore: get rid off statfs update on each txc, the rest
of the PRs are Rados approved."

**Comment by ljflores** [comment]
Rados suite review: http://pulpito.front.sepia.ceph.com/?branch=wip-yuri7-testing-2022-10-17-0814

Failures:, unrelated
    1. https://tracker.ceph.com/issues/57311
    2. https://tracker.ceph.com/issues/52321
    3. https://tracker.ceph.com/issues/52657
    4. https://tracker.ceph.com/issues/57935

Details:
    1. rook: ensure CRDs are installed first - Ceph - Orchestrator
    2. qa/tasks/rook times out: 'check osd count' reached maximum tries (90) after waiting for 900 seconds - Ceph - Orchestrator
    2. rados/thrash-erasure-code: wait_for_recovery timeout due to "active+clean+remapped+laggy" pgs - Ceph - RADOS
    3. MOSDPGLog::encode_payload(uint64_t): Assertion `HAVE_FEATURE(features, SERVER_NAUTILUS)' - Ceph - RADOS
    4. all test jobs get stuck at "Running task ansible.cephlab..." - Infrstructure - Sepia

**Pull request merged**